macros/stage_test_failures.sql

```diff
@@ -8,11 +8,20 @@
     {%- for result in results -%}
       {%- if result.node.resource_type == 'test' -%}
         {%- if result.status == 'error' or result.status == 'fail' -%}
-          {%- do failed_tests.append({
-            'name': result.node.name,
-            'unique_id': result.node.unique_id,
-            'failures_query': 'select * from ' ~ result.node.relation_name
-          }) -%}
+          -- staging failures only works if the test is configured to store failues — we depend on the ephemeral table referenced
+          -- in the `failures_query` below
+          {%- if result.node.config.get('store_failures', False) == True or (should_store_failures() and result.node.config.get('store_failures', False) != False) -%}
+            {%- set metaplane_meta = result.node.meta.get("metaplane", { 'stage_failures': false }) -%}
```

Comment:
🤔  I wonder if we should have a name a little more general than "stage_failures". Maybe it should be more direct like "downloadable_failures" or "shareable_failures".